Assets/MixedRealityToolkit.Tests/PlayModeTests/PlayModeTestUtilities.cs

```diff
@@ -1,23 +1,82 @@
+using System.Collections;
 
 namespace Microsoft.MixedReality.Toolkit.Tests
 {
     public class PlayModeTestUtilities
     {
-        public static SimulatedHandData.HandJointDataGenerator GenerateHandPose(ArticulatedHandPose.GestureId gesture, Handedness handedness, Vector3 screenPosition)
+        public static SimulatedHandData.HandJointDataGenerator GenerateHandPose(ArticulatedHandPose.GestureId gesture, Handedness handedness, Vector3 worldPosition)
         {
             return (jointsOut) =>
             {
                 ArticulatedHandPose gesturePose = ArticulatedHandPose.GetGesturePose(gesture);
                 Quaternion rotation = Quaternion.identity;
-                Vector3 position = CameraCache.Main.ScreenToWorldPoint(screenPosition);
-                gesturePose.ComputeJointPoses(handedness, rotation, position, jointsOut);
+                gesturePose.ComputeJointPoses(handedness, rotation, worldPosition, jointsOut);
             };
         }
+
+        public static IMixedRealityInputSystem GetInputSystem()
+        {
+            IMixedRealityInputSystem inputSystem;
+            MixedRealityServiceRegistry.TryGetService(out inputSystem);
+            Assert.IsNotNull(inputSystem, "MixedRealityInputSystem is null!");
+            return inputSystem;
+        }
+
+        public static InputSimulationService GetInputSimulationService()
+        {
+            IMixedRealityInputSystem inputSystem = GetInputSystem();
+            InputSimulationService inputSimulationService = (inputSystem as IMixedRealityDataProviderAccess).GetDataProvider<InputSimulationService>();
+            Assert.IsNotNull(inputSimulationService, "InputSimulationService is null!");
+            inputSimulationService.UserInputEnabled = false;
+            return inputSimulationService;
+        }
+
+        internal static IEnumerator MoveHandFromTo(Vector3 startPos, Vector3 endPos, int numSteps, ArticulatedHandPose.GestureId gestureId, Handedness handedness, InputSimulationService inputSimulationService)
+        {
+            for (int i = 0; i < numSteps; i++)
+            {
+                float t = 1.0f / numSteps * i;
+                Vector3 handPos = Vector3.Lerp(startPos, endPos, t);
+                var handDataGenerator = GenerateHandPose(
+                        gestureId,
+                        handedness,
+                        handPos);
+                Debug.Assert(handedness == Handedness.Right || handedness == Handedness.Left, "handedness must be either right or left");
+                SimulatedHandData toUpdate = handedness == Handedness.Right ? inputSimulationService.HandDataRight : inputSimulationService.HandDataLeft;
+                bool isPinching = gestureId == ArticulatedHandPose.GestureId.Grab || gestureId == ArticulatedHandPose.GestureId.Pinch || gestureId == ArticulatedHandPose.GestureId.PinchSteadyWrist;
+                inputSimulationService.HandDataRight.Update(true, false, handDataGenerator);
```

Comment:
this should actually use isPinching

Reply:
Yes, the way we determine if simulated hands are pinching/grabbing is not great atm (see also [SimulatedHandState.GestureBlending](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/ae0b1467b564c5b647c5190c84fe725b8ffd6f6a/Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedHandDataProvider.cs#L64). I'm not sure of the best way to tag hand poses for these actions, perhaps add a code comment so we improve it in both places.